S3CSI-78: Improve documentation UX for end users

anurag4DSB opened the pull request:
Note to reviewers: 1st commit will be deleted, its only for demo for the [website changes](https://scality.github.io/mountpoint-s3-csi-driver/)

Changes:
- Change theme switcher icons
- Add Scality Logo to site and favicon
- Added custom css with below changes

Light theme update:
Old version:
<img width="2032" alt="Screenshot 2025-06-05 at 09 19 49" src="https://github.com/user-attachments/assets/4251582a-7a0e-49ef-8a36-8d2ec44bb085" />


New version: 
<img width="2032" alt="Screenshot 2025-06-05 at 11 10 36" src="https://github.com/user-attachments/assets/42578999-2ccc-411d-b202-e014e2c4d475" />

Dark theme update:
Old version:
<img width="2032" alt="Screenshot 2025-06-05 at 09 20 26" src="https://github.com/user-attachments/assets/b0a077aa-979d-40d8-8410-a2dafedb7008" />

New version: 
<img width="2032" alt="Screenshot 2025-06-05 at 11 09 55" src="https://github.com/user-attachments/assets/616d5e0d-0569-4301-b444-75f6d8ea3e57" />

dvasilas reviewed (approved):
I have limited css knowledge, but it looks nice so :heavy_check_mark: 

My only minor comment is that in the dark mode the horizontal bar at the top is almost the same color as the rest of the page (but I think it's not an issue).

anurag4DSB commented:
> > I have limited css knowledge, but it looks nice so ✔️
> > My only minor comment is that in the dark mode the horizontal bar at the top is almost the same color as the rest of the page (but I think it's not an issue).
> 
> Thanks for raising that, I updated header shadows. Everything is possible with CSS :)

I changed it to a line now for both modes, as show was too ugly on dark theme.

anurag4DSB commented:
Just added 2 commits for changing accent colors and making sure search is actually following the theme enabled by end user
